### PR TITLE
chore: stabilize SSE tests

### PR DESCRIPTION
**Pull request by dupdob**
Clean up the design of SSE server tests so that they are no longer flaky

### CODE REVIEW

**Comment by rouke-broersma** [comment]
@xandervedder could you review this?

**Comment by dupdob** [comment]
crap: more stable, but still not fully stable

**Comment by dupdob** [comment]
I increased the timeouts, to cover risks for slower build machines.

File: src/Stryker.Core/Stryker.Core.UnitTest/Reporters/Html/Realtime/SseServerTest.cs [hunk 1]
```diff
@@ -1,21 +1,54 @@
...
+    private bool WaitConnection(int timeout)
+    {
+        var watch = new Stopwatch();
+        watch.Start();
+        lock (_lock)
+        {
+            while (!_connected && watch.ElapsedMilliseconds < timeout)
+            {
+                Monitor.Wait(_lock, Math.Max(timeout - (int)watch.ElapsedMilliseconds, 1));
+            }
```
Comment (xandervedder) [inline src/Stryker.Core/Stryker.Core.UnitTest/Reporters/Html/Realtime/SseServerTest.cs:44]: Is the while loop necessary? What if we only do this:

```suggestion
            Monitor.Wait(_lock, Math.Max(timeout, 1));
```

Should be functionally equivalent, no?

File: src/Stryker.Core/Stryker.Core.UnitTest/Reporters/Html/Realtime/SseServerTest.cs [hunk 2]
```diff
@@ -1,21 +1,54 @@
...
+    private void ClientConnected(object sender, EventArgs e)
+    {
+        lock (_lock)
+        {
+            _connected = true;
+            Monitor.Pulse(_lock);
+        }
+    }
+
+    private bool WaitConnection(int timeout)
```
Comment (xandervedder) [inline src/Stryker.Core/Stryker.Core.UnitTest/Reporters/Html/Realtime/SseServerTest.cs:35]:
```suggestion
    private bool WaitForConnection(int timeout)
```

Comment (dupdob) [reply src/Stryker.Core/Stryker.Core.UnitTest/Reporters/Html/Realtime/SseServerTest.cs]: done

**Pull request merged**